devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy

```diff
@@ -0,0 +1,171 @@
+        capability "Actuator"
+        capability "Button"
+        capability "Battery"
+        capability "Holdable Button"
+        capability "Configuration"
+        capability "Sensor"
+        capability "Health Check"
+
+        fingerprint mfr: "0086", model: "0082", deviceJoinName: "Aeotec Wallmote Quad", mnmn: "SmartThings", vid: "generic-4-button"
+        fingerprint mfr: "0086", model: "0081", deviceJoinName: "Aeotec Wallmote", mnmn: "SmartThings", vid: "generic-2-button"
+    }
+
+    tiles(scale: 2) {
+        multiAttributeTile(name: "rich-control", type: "generic", width: 6, height: 4, canChangeIcon: true) {
+            tileAttribute("device.button", key: "PRIMARY_CONTROL") {
+                attributeState "default", label: ' ', action: "", icon: "st.unknown.zwave.remote-controller", backgroundColor: "#ffffff"
+            }
+        }
+        valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "battery", label:'${currentValue}% battery', unit:""
+        }
+
+        main("rich-control")
+        details(["rich-control", childDeviceTiles("endpoints"), "battery"])
+    }
+}
+
+def getNumberOfButtons() {
+    def modelToButtons = ["0082" : 4, "0081": 2]
+    return modelToButtons[zwaveInfo.model] ?: 1
+}
+
+def installed() {
+    sendEvent(name: "numberOfButtons", value: numberOfButtons)
+    sendEvent(name: "DeviceWatch-Enroll", value: [protocol: "zwave", scheme:"untracked"].encodeAsJson(), displayed: false)
+    sendEvent(name: "button", value: "pushed", data: [buttonNumber: 1])
+    createChildDevices()
+    response([
+            secure(zwave.batteryV1.batteryGet()),
+            "delay 2000",
+            secure(zwave.wakeUpV2.wakeUpNoMoreInformation())
+    ])
+}
+
+def updated() {
+    if (!childDevices) {
+        createChildDevices()
+    }
+    else if (device.label != state.oldLabel) {
+        childDevices.each {
+            def segs = it.deviceNetworkId.split(":")
+            def newLabel = "${device.displayName} button ${segs[-1]}"
+            it.setLabel(newLabel)
+        }
+        state.oldLabel = device.label
+    }
+}
+
+def configure() {
+
+}
+
+def parse(String description) {
+    def results = []
+    if (description.startsWith("Err")) {
+        results = createEvent(descriptionText:description, displayed:true)
+    } else {
+        def cmd = zwave.parse(description)
+        if(cmd) results += zwaveEvent(cmd)
+        if(!results) results = [ descriptionText: cmd, displayed: false ]
+    }
+    return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.centralscenev1.CentralSceneNotification cmd) {
+    def button = cmd.sceneNumber
+    def value
+    // 0 = pushed, 1 = released, 2 = held down
+    if (cmd.keyAttributes != 2) {
+        value = cmd.keyAttributes == 1 ? "held" : "pushed"
+    } else {
+        // we can't do anything with the held down event yet
```

Comment:
why not?

Reply:
It's sent repeatedly and just signifies that the button is still held.

Reply:
Hmmmm createEvent should see that the state for the button attribute hasn't changed so it will set the stateChanged flag accordingly and only the first "held" will pass as an event. So it should be safe to send pushed on 0 and held on 2.